### ISSUE TITLE
🐛 [Bug]: Unclear "json: invalid use of ,string struct tag, trying to unmarshal unquoted value into uint64"

### ISSUE THREAD

**Issue by kokizzu** [gofiber/fiber#3025]
### Bug Description

Currently it doesn't show which struct have that are uint64 that cannot be parsed by BodyParser

### How to Reproduce

1. Create a struct with multiple field like this: uint64 `json:"fieldNameXX,string"` field, an endpoint, call BodyParser with payload unquoted an quoted
2. Got those error

### Expected Behavior

It would show which field failed to be parsed

for quality of life it should be more permissive between quoted and unquoted number

### Fiber Version

v2.52.4

### Code Snippet (optional)

```go
package main

import (
	"log"

	"github.com/gofiber/fiber/v2"
)

func main() {
	app := fiber.New()

	type Foo struct {
		A uint64 `json:"a,string"`
		B int64  `json:"b,string"`
		C uint64 `json:"c"`
		D int64  `json:"d"`
	}

	app.Post("/", func(c *fiber.Ctx) error {
		var f Foo
		err := c.BodyParser(&f)
		if err != nil {
			return err
		}
		return c.JSON(f)
	})

	log.Fatal(app.Listen(":3000"))
}

/*
curl localhost:3000 -X POST --data '{"a":123,"b":"123","c":"123","d":123}' -H "Content-Type: application/json"
json: invalid use of ,string struct tag, trying to unmarshal unquoted value into int64%
*/
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by renanbastos93** [comment]
@kokizzu It doesn't sound like a bug because your JSON is wrong. I've tested using `json.Unmarshal` from Go.
see below
```go
package main

import (
	"encoding/json"
	"fmt"
)

type Foo struct {
	A uint64 `json:"a,string"`
	B int64  `json:"b,string"`
	C uint64 `json:"c"`
	D int64  `json:"d"`
}

func main() {
	var f Foo
	// json: invalid use of ,string struct tag, trying to unmarshal unquoted value into uint64 {0 123 0 123}
	err := json.Unmarshal([]byte(`{"a":123,"b":"123","c":"123","d":123}`), &f)
	fmt.Println(err, f)

	// success
	err = json.Unmarshal([]byte(`{"a":"123","b":"123","c":123,"d":123}`), &f)
	fmt.Println(err, f)
}
```

**Comment by kokizzu** [comment]
yes, what i'm asking is clearer error message

**Comment by kokizzu** [comment]
I mean if i have 24 fields, i didn't know which one is the issue, this one is just minimal example of 4 fields, i know for sure what's wrong for this one, but if there's 24 fields, that's pain

**Comment by ReneWerner87** [comment]
@kokizzu We use the encoder or decoder of the standard packet internally
This could be improved 
Can you create an issue report in the golang main repository

**Comment by kokizzu** [comment]
apparently this related
https://github.com/golang/go/issues/34472

**Comment by renanbastos93** [comment]
@kokizzu Of course, but it isn't a bug from Fiber. I've been trying to understand where this error message is concatenated, but I haven't found it yet. 
According to @ReneWerner87 , this problem is on Golang's main repository.

**Comment by renanbastos93** [comment]
Hey @kokizzu, I apologize if my communication hasn't been very clear. I think I understood your suggestion. If you need help contributing to the Golang main repository, just let me know.

**Comment by kokizzu** [comment]
no problem, end up using json5 decoder with number-string patch instead
